workspace-manager-clienttests/src/main/java/scripts/utils/GcpCloudContextTestScriptBase.java

```diff
@@ -0,0 +1,79 @@
+package scripts.utils;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+import bio.terra.testrunner.runner.config.TestUserSpecification;
+import bio.terra.workspace.api.WorkspaceApi;
+import bio.terra.workspace.client.ApiException;
+import bio.terra.workspace.model.CloudPlatform;
+import bio.terra.workspace.model.CreateCloudContextRequest;
+import bio.terra.workspace.model.CreateCloudContextResult;
+import bio.terra.workspace.model.JobControl;
+import bio.terra.workspace.model.JobReport.StatusEnum;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Fixture for tests that use a single workspace with a GCP cloud context as a fixture. The
+ * expectation is that the workspace and context setup & deletion are all that needs to happen in
+ * setup() and cleanup(), although doSetup() and doCleanup() can still be overridden, provided the
+ * caller calls super().
+ *
+ * <p>Per WorkspaceAllocateTestScriptBase, you must specify the spend profile to use as the first
+ * parameter in the test configuration.
+ *
+ * <p>No doUserJourney() implementation is provided, and this must be overridden by inheriting
+ * classes.
+ */
+public abstract class GcpCloudContextTestScriptBase extends WorkspaceAllocateTestScriptBase {
```

Comment:
Using inheritance to do test setup worries me. Java doesn't support multiple inheritance, so if we have two setup cases, and want to combine them, we have to start making multiple classes. Can we use composition instead of inheritance here to do various bits of setup? The composable setup bits can have input parameters and accessor functions for the things they create, like the project id. 
Not necessarily this PR to change everything, but maybe at least start a better way for this new setup wrinkle.

Reply:
At least so far, the setup cases have been pretty distinct and broad (workspace fixture -> cloud context fixture) - I don't anticipate building more of these fixtures in the near future. I think we probably should move to composition if we expand the set more, but as-is I'm not sure when we'd do that.

Reply:
looks like I might not be the one starting anyways - #274 introduced a static method way of doing it, pulled that in here to hopefully minimize merge conflicts